backends/cuda/ceed-cuda-operator.c

```diff
@@ -212,17 +245,25 @@ static int CeedOperatorApplyAdd_Cuda(CeedOperator op, CeedVector invec,
     } else {
       // Get input vector
       ierr = CeedOperatorFieldGetVector(opinputfields[i], &vec); CeedChk(ierr);
-      if (vec == CEED_VECTOR_ACTIVE)
-        vec = invec;
-      // Restrict
+      // Get input element restriction
       ierr = CeedOperatorFieldGetElemRestriction(opinputfields[i], &Erestrict);
       CeedChk(ierr);
-      ierr = CeedElemRestrictionApply(Erestrict, CEED_NOTRANSPOSE, vec,
+      if (vec == CEED_VECTOR_ACTIVE) 
+        vec = invec;
+      // Restrict, if necessary
+      if (impl->evecs[i] == NULL) {
```

Comment:
or `!impl->evecs[i]` if you prefer?

Reply:
Yes, I agree that's cleaner, I changed it.

Reply:
It is classically not considered cleaner though.

Reply:
@YohannDudouit do you want me to change it back?  I don't have strong feelings either way.

Reply:
No idea what "classically" means here. PETSc style uses `if (!obj)` and we use that pattern plenty in libCEED.  Whatever you choose, let's not bikeshed this.

Reply:
Let's leave it as it is.  I think the version @jeremylt suggested fits in better with the rest of libCEED.

Reply:
FWIW, I'm one of those who think `if (impl->evecs[i] == NULL)` is easier to read, but if the other way is preferred for consistency with the rest of the library, it makes sense to be consistent. Just saying.. Not so important